sleap/gui/widgets/video.py

```diff
@@ -1732,6 +1733,211 @@ def updateEdge(self, node: QtNode):
             self.setLine(line)
 
 
+class VisibleBoundingBox(QtWidgets.QGraphicsRectItem):
+    """
+    QGraphicsRectItem for user instance bounding boxes.
+
+    This object defines a scalable bounding box that encases an instance and handles
+    the relevant scaling operations. It is instantiated when its respective QtInstance
+    object is instantiated.
+
+    When instantiated, it creates 4 boxes, which are properties of the overall object,
+    on the corners of the overall bounding box. These corner boxes can be dragged to
+    scale the overall bounding box.
+
+    Args:
+        rect: The :class:`QRectF` object which defines the non-scalable bounding box.
+        parent: The :class:`QtInstance` to encompass.
+
+    """
+
+    def __init__(self, rect, parent=None):
+        super().__init__(rect, parent)
+        if parent is not None:
+            self.box_width = parent.markerRadius
+            color_manager = parent.player.color_manager
+            int_color = color_manager.get_item_color(parent.instance)
+            self.int_color = QColor(*int_color)
+        else:
+            self.box_width = 7
+            self.int_color = Qt.white
```

Comment:
Is there ever a case when `parent is None` - do we ever want `parent` to be `None`? Maybe we should change the optional keyword argument to a required positional argument and get rid of this conditional.
```suggestion
    def __init__(self, rect, parent):
        super().__init__(rect, parent)
        self.box_width = parent.markerRadius
        color_manager = parent.player.color_manager
        int_color = color_manager.get_item_color(parent.instance)
        self.int_color = QColor(*int_color)
```

Reply:
done!